saleor/webhook/transport/tests/test_utils.py

```diff
@@ -74,3 +78,90 @@ def test_webhook_retry_redirect(webhook, event_delivery):
     # then
     retry = handle_webhook_retry(task, webhook, response, event_delivery, attempt)
     assert retry is False
+
+
+@pytest.mark.parametrize(
+    (
+        "target_url",
+        "expected_access_key_id",
+        "expected_secret_access_key",
+        "expected_region",
+        "is_fifo",
+        "expected_queue_url",
+    ),
+    [
+        (
+            "awssqs://key_id:secret%2B%2Faccess@sqs.eu-west-1.amazonaws.com/account_id/queue_name",
+            "key_id",
+            "secret+/access",
+            "eu-west-1",
+            False,
+            "https://sqs.eu-west-1.amazonaws.com/account_id/queue_name",
+        ),
+        (
+            "awssqs://key_id:secret@sqs.example.com/account_id/queue_name.fifo",
+            "key_id",
+            "secret",
+            "us-east-1",
+            True,
+            "https://sqs.example.com/account_id/queue_name.fifo",
+        ),
+    ],
+)
+def test_send_webhook_using_aws_sqs(
+    target_url,
+    expected_access_key_id,
+    expected_secret_access_key,
+    expected_region,
+    is_fifo,
+    expected_queue_url,
+    monkeypatch,
+):
+    # given
+    message, signature, domain = "payload", "signature", "example.com"
+    event_type = WebhookEventAsyncType.ORDER_CREATED
+    sqs_response = {
+        "MD5OfMessageBody": "message-body-hash",
+        "MD5OfMessageAttributes": "message-attributes-hash",
+        "MD5OfMessageSystemAttributes": "message-system-attributes-hash",
+        "MessageId": "message-id",
+        "SequenceNumber": "sequence-number",
+    }
+    mocked_client = MagicMock()
+    mocked_client.send_message.return_value = sqs_response
+    mocked_client_constructor = MagicMock(spec=boto3.client, return_value=mocked_client)
+    monkeypatch.setattr(
+        "saleor.webhook.transport.utils.boto3.client",
+        mocked_client_constructor,
+    )
+
+    # when
+    webhook_response = send_webhook_using_aws_sqs(
+        target_url, message.encode("utf-8"), domain, signature, event_type
+    )
+
+    # then
+    mocked_client_constructor.assert_called_once_with(
+        "sqs",
+        region_name=expected_region,
+        aws_access_key_id=expected_access_key_id,
+        aws_secret_access_key=expected_secret_access_key,
+    )
+    expected_call_args = {
+        "QueueUrl": expected_queue_url,
+        "MessageAttributes": {
+            "SaleorDomain": {"DataType": "String", "StringValue": domain},
+            "SaleorApiUrl": {
+                "DataType": "String",
+                "StringValue": f"http://{domain}/graphql/",
+            },
+            "EventType": {"DataType": "String", "StringValue": event_type},
+            "Signature": {"DataType": "String", "StringValue": signature},
+        },
+        "MessageBody": message,
+    }
+    if is_fifo:
```

Comment:
As a rule of thumb if you need to use an `if` statement in your tests this means you should split the test into more than one. In your case `parametrize` is used to incorporate the logic that should be represented in the name/body of the test and the single True/False in the parametrize statement is an additional overhead someone will have to take on when they read the test. So this test should be split into two, one for fifo and one without.

Reply:
Great point, thank you. I've split this test as suggested